src/modules/partition/jobs/CreatePartitionJob.cpp

```diff
@@ -32,9 +33,134 @@ CreatePartitionJob::CreatePartitionJob( Device* device, Partition* partition )
+    { "b921b045-1df0-41c3-af44-4c6f280d3fae", "Root Partition (64-bit ARM)" },
+    { "993d8d3d-f80e-4225-855a-9daf8ed7ea97", "Root Partition (Itanium/IA-64)" },
+    { "d13c5d3b-b5d1-422a-b29f-9454fdc89d76", "Root Verity Partition (x86)" },
+    { "2c7357ed-ebd2-46d9-aec1-23d437ec2bf5", "Root Verity Partition (x86-64)" },
+    { "7386cdf2-203c-47a9-a498-f2ecce45a2d6", "Root Verity Partition (32-bit ARM)" },
+    { "df3300ce-d69f-4c92-978c-9bfb0f38d820", "Root Verity Partition (64-bit ARM/AArch64)" },
+    { "86ed10d5-b607-45bb-8957-d350f23d0571", "Root Verity Partition (Itanium/IA-64)" },
+    { "75250d76-8cc6-458e-bd66-bd47cc81a812", "/usr Partition (x86)" },
+    { "8484680c-9521-48c6-9c11-b0720656f69e", "/usr Partition (x86-64)" },
+    { "7d0359a3-02b3-4f0a-865c-654403e70625", "/usr Partition (32-bit ARM)" },
+    { "b0e01050-ee5f-4390-949a-9101b17104e9", "/usr Partition (64-bit ARM/AArch64)" },
+    { "4301d2a6-4e3b-4b2a-bb94-9e0b2c4225ea", "/usr Partition (Itanium/IA-64)" },
+    { "8f461b0d-14ee-4e81-9aa9-049b6fb97abd", "/usr Verity Partition (x86)" },
+    { "77ff5f63-e7b6-4633-acf4-1565b864c0e6", "/usr Verity Partition (x86-64)" },
+    { "c215d751-7bcd-4649-be90-6627490a4c05", "/usr Verity Partition (32-bit ARM)" },
+    { "6e11a4e7-fbca-4ded-b9e9-e1a512bb664e", "/usr Verity Partition (64-bit ARM/AArch64)" },
+    { "6a491e03-3be7-4545-8e38-83320e0ea880", "/usr Verity Partition (Itanium/IA-64)" },
+    { "933ac7e1-2eb4-4f13-b844-0e14e2aef915", "Home Partition" },
+    { "3b8f8425-20e0-4f3b-907f-1a25a76f98e8", "Server Data Partition" },
+    { "4d21b016-b534-45c2-a9fb-5c16e091fd2d", "Variable Data Partition" },
+    { "7ec6f557-3bc5-4aca-b293-16ef5df639d1", "Temporary Data Partition" },
+    { "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f", "Swap" },
+    { "c12a7328-f81f-11d2-ba4b-00a0c93ec93b", "EFI System Partition" },
+    { "bc13c2ff-59e6-4262-a352-b275fd6f7172", "Extended Boot Loader Partition" },
+    { "0fc63daf-8483-4772-8e79-3d69d8477de4", "Other Data Partitions" },
+    { "ebd0a0a2-b9e5-4433-87c0-68b6b72699c7", "Microsoft basic data" },
+};
+
+static QString
+prettyGptType( const QString& type )
+{
+    return gptTypePrettyStrings.value( type.toLower(), type );
+}
+
+static QString
+prettyGptAttributes( quint64 attributes )
+{
+    QStringList list;
+
+    if ( attributes & ( 1UL << 0 ) )
+    {
+        list += QString( "RequiredPartition" );
+    }
+
+    if ( attributes & ( 1UL << 1 ) )
+    {
+        list += QString( "NoBlockIOProtocol" );
+    }
+
+    if ( attributes & ( 1UL << 2 ) )
+    {
+        list += QString( "LegacyBIOSBootable" );
+    }
+
+    for ( qint8 bit = 3; bit < 64; bit++ )
+    {
+        if ( attributes & ( 1UL << bit ) )
+        {
+            list += QString( "Bit" ) + QString::number( bit );
+        }
+    }
+
+    QString string = list.join( QStringLiteral( " and " ) );
+    if ( string.size() > 64 )   
+    {
+        return QString( "0x" ) + QString::number( attributes, 0x10 );
```

Comment:
This seems a bit dramatic: perhaps keep the human-readable bits and hex the rest, rather than **also** dumping the human-readable part overboard?

Reply:
Yep, this is another solution. I have not make my mind yet.

Normally no one use these bits, at least for now and it concerns only the users of `partitionLayout`.

There are only 2 bits (+ 1 for ESP) used by [systemd](https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html) (see Table 2. Partition Attributes):

Name | Value | Applicable to | Explanation
-- | -- | -- |  --
GPT_FLAG_READ_ONLY | 0x1000000000000000 | /, /home/, /srv/, /var/, /var/tmp/, Extended Boot Loader Partition | Partition is mounted read-only
GPT_FLAG_NO_AUTO | 0x8000000000000000 | /, /home/, /srv/, /var/, /var/tmp/, Extended Boot Loader Partition | Partition is not mounted automatically
GPT_FLAG_NO_BLOCK_IO_PROTOCOL | 0x0000000000000002 | EFI System Partition (ESP) | Partition is not mounted automatically

4 bits used by [Microsoft](https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_entries_(LBA_2%E2%80%9333)) to define the type-specific attributes for basic data partition, aka `EBD0A0A2-B9E5-4433-87C0-68B6B72699C7` (See Partition entries (LBA 2–33), Basic data partition attributes):

Bit | Content
-- | --
60 | Read-only
61 | Shadow copy (of another partition)
62 | Hidden
63 | No drive letter (i.e. do not automount)

And there are 9 others bits used by [Google](https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_entries_(LBA_2%E2%80%9333)) to define the type-specific attributes for Chrome OS kernel (See Partition entries (LBA 2–33), Chrome OS kernel partition attributes):

Bit | Content
-- | --
56 | Successful boot flag
55–52 | Tries remaining
51–48 | Priority (15: highest, 1: lowest, 0: not bootable)